mountpoint-s3-fs/src/metrics.rs

```diff
@@ -30,8 +38,13 @@ pub const TARGET_NAME: &str = "mountpoint_s3_fs::metrics";
 /// done with their work; metrics generated after shutting down the sink will be lost.
 ///
 /// Panics if a sink has already been installed.
-pub fn install() -> MetricsSinkHandle {
-    let sink = Arc::new(MetricsSink::new());
+pub fn install(
+    #[cfg(feature = "otlp_integration")] otlp_config: Option<OtlpConfig>,
```

Comment:
Instead of having the feature define if the argument is allowed, perhaps we could change the type signature from `Option<OtlpConfig>` to `Option<SomethingElse>`. Perhaps an enum with no values? That would allow callers to keep the same arguments (None) in all the cases you've special cased for now